main.cpp

```diff
@@ -210,285 +239,379 @@ struct Museum
+{
+    if (thickness == 0.75f)
+    {
+        plywoodInStock += (sheets * 8.f);
+    }
+}
+
+bool SubwooferFactory::sellSubwoofer(float price)
+{
+    if(completedSubwoofers > 0 && price > 1.f)
+    {
+        completedSubwoofers--;
+        return true;
+    }
+    return false;
+}
+
 struct FreightTrain
 {
-    // 1) the number of locomotives (int)
     int numberOfLocomotives = 1;
-    // 2) the number of freight cars (int)
     int numberOfFreightCars = 1;
-    // 3) the maximum cargo weight able to be pulled (float)
     float maxCargoWeight { numberOfLocomotives * 8000.f };
-    // 4) the gross cargo weight in tons (float)
     float grossCargoWeight { numberOfFreightCars * 110.f };
-    // 5) the conductors name (std::string)
     std::string conductorName = "conductor";
 
-    // 1) proceed to next stop
     void proceedToNextStop();
-    // 2) pickup / dropoff cars
-    void pickupOrDropoffCars(int numOfCars)
-    {
-        numberOfFreightCars += numOfCars;
-    }
-    // 3) blow airhorn
+    void pickupOrDropoffCars(int numOfCars);
     void blowAirHorn(float durationInSeconds);
 };
-/*
-Thing 5) Power Supply
-5 properties:
-    1) Mains Input voltage (float)
-    2) Main Output Voltage (Vcc) (float)
-    3) Heater Output Voltage (float)
-    4) Maximum Current (float)
-    5) Fuse state (bool)
-3 things it can do:
-    1) The mains input voltage can be switched on / off
-    2) the Main output voltage can be switched on independently (standby)
-    3) The mains fuse can blow
-*/
+
+void FreightTrain::proceedToNextStop()
+{
+    blowAirHorn(5.f);
+}
+
+void FreightTrain::pickupOrDropoffCars(int numOfCars)
+{
+    numberOfFreightCars += numOfCars;
+    if (numberOfFreightCars < 0)
+        numberOfFreightCars = 0;
+}
+
+void FreightTrain::blowAirHorn(float durationInSeconds)
+{
+    if (durationInSeconds > 0.f)
+    {
+        // do something noisy for durationInSeconds
+    }
+}
+
 struct PowerSupply
 {
-    // 1) Mains Input voltage (float)
-    float mainsInputVoltage = 110.f;
-    // 2) Main Output Voltage (Vcc) (float)
-    float mainOutputVoltage = 450.f;
-    // 3) Heater Output Voltage (float)
-    float heaterOutputVoltage = 6.3f;
-    // 4) Maximum Current (float)
+    float mainsInputVoltage = 115.f;
+    float mainOutputVoltage = 460.f;
+    float heaterOutputVoltage = 6.325f;
     float maximumCurrentInAmps = 1.f;
-    // 5) Fuse state (bool)
     bool fuseState { true };
 
-    // 1) The mains input voltage can be switched on / off
     void setPowerState(bool powerState);
-    // 2) the Main output voltage can be switched on independently (standby)
     void setStandbyState(bool standbyState);
-    // 3) The mains fuse can blow
-    void blowFuse() { fuseState = false;}
+    void blowFuse(); //{ fuseState = false;}
+private:
+    bool _powerState = 1;
+    bool _standbyState = 1;
+    float powerTransformerMainRatio = 4.f;
+    float powerTransformerHeaterRatio = 0.055f;
```

Comment:
don't use `private`.  private is covered in Project 7
Don't use underscores in member variable names
- [ ] revise